client/extensions/hello-dolly/state/reducer.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Internal dependencies
+ */
+import {
+	HELLO_DOLLY_NEXT_LYRIC,
+} from './action-types';
+
+const lyrics = [
+	'Hello, Dolly',
+	'Well, hello, Dolly',
+	'It\'s so nice to have you back where you belong',
+	'You\'re lookin\' swell, Dolly',
+	'I can tell, Dolly',
+	'You\'re still glowin\', you\'re still crowin\'',
+	'You\'re still goin\' strong',
+	'We feel the room swayin\'',
+	'While the band\'s playin\'',
+	'One of your old favourite songs from way back when',
+	'So, take her wrap, fellas',
+	'Find her an empty lap, fellas',
+	'Dolly\'ll never go away again',
+	'Hello, Dolly',
+	'Well, hello, Dolly',
+	'It\'s so nice to have you back where you belong',
+	'You\'re lookin\' swell, Dolly',
+	'I can tell, Dolly',
+	'You\'re still glowin\', you\'re still crowin\'',
+	'You\'re still goin\' strong',
+	'We feel the room swayin\'',
+	'While the band\'s playin\'',
+	'One of your old favourite songs from way back when',
+	'Golly, gee, fellas',
+	'Find her a vacant knee, fellas',
+	'Dolly\'ll never go away',
+	'Dolly\'ll never go away',
+	'Dolly\'ll never go away again'
+];
+
+const initialState = {
+	lyric: lyrics[ 0 ],
```

Comment:
it would be nice for example code to illustrate the concept of storing the minimum necessary state to represent the app. in this case, why do we need to store the lyric? do we plan on adding lyrics at runtime? if not, why not just store the index and allow some selector to pull out the text of the referenced lyrics?

keeping state access indirect will also give us flexibility as we grow, so might be good for example code